libraries/chain/db_maint.cpp

```diff
@@ -442,7 +440,7 @@ void database::initialize_budget_record( fc::time_point_sec now, budget_record&
    budget_u128 += ((uint64_t(1) << GRAPHENE_CORE_ASSET_CYCLE_RATE_BITS) - 1);
    budget_u128 >>= GRAPHENE_CORE_ASSET_CYCLE_RATE_BITS;
    if( budget_u128 < reserve.value )
```

Comment:
This generates a signed/unsigned comparison warning on gcc 7.4.0, not the fault of this PR though.